### PR TITLE
🖍️  improving type-hints in `speechbrain/pretrained/interfaces.py`

### PR DESCRIPTION
**Pull request by jonasvdd**
It is not a big PR, but at least it is honest work 😅 

This PR:
* adjusts the wrongfully typed `boundary` argument (was set to `str` type, but must be `torch.Tensor`).
* homogenizes the tensor type-hints to be all of the type `torch.Tensor` which is more correct. See 📷 ⬇️ 

image

### CODE REVIEW

File: speechbrain/pretrained/interfaces.py [hunk 1]
```diff
@@ -2724,7 +2724,7 @@ def encode_batch(self, texts):
         with torch.no_grad():
             inputs = [
                 {
-                    "text_sequences": torch.tensor(
+                    "text_sequences": torch.Tensor(
```
Comment (jonasvdd) [inline speechbrain/pretrained/interfaces.py:2727]: Whoops. Accidentally altered this one. Will fix it in a new commit.

Comment (jonasvdd) [reply speechbrain/pretrained/interfaces.py]: Should be fixed now!

**Comment by jonasvdd** [comment]
In https://github.com/speechbrain/speechbrain/pull/1725/commits/f372dffd6ed141d6a041cea4637a3dc8967186ae, I fix some more typos throughout the whole interfaces.py file! 😄

**Review by Adel-Moumen — APPROVED** [review]
LGTM!

Thanks a lot for the PR, you did a great job! I wish you an happy new year :-)

**Comment by jonasvdd** [comment]
@Adel-Moumen

Happy holidays for you as well! 🎉

**Pull request merged**